.vscode/iTwinUISnippets.code-snippets

```diff
@@ -0,0 +1,63 @@
+{
+  // Place your iTwinUI workspace snippets here. Each snippet is defined under a snippet name and has a scope, prefix, body and
+  // description. Add comma separated ids of the languages where the snippet is applicable in the scope field. If scope
+  // is left empty or omitted, the snippet gets applied to all languages. The prefix is what is
+  // used to trigger the snippet and the body will be expanded and inserted. Possible variables are:
+  // $1, $2 for tab stops, $0 for the final cursor position, and ${1:label}, ${2:another} for placeholders.
+  // Placeholders with the same ids are connected.
+  // Example:
+  // "Print to console": {
+  // 	"scope": "javascript,typescript",
+  // 	"prefix": "log",
+  // 	"body": [
+  // 		"console.log('$1');",
+  // 		"$2"
+  // 	],
+  // 	"description": "Log output to console"
+  // }
+  "Separator": {
+    "prefix": "//-",
+    "body": [
+      "// ------------------------------------------------------------------------------------------------",
+      "$1"
+    ],
+    "description": "Creates a separator comment"
+  },
+  "Component": {
+    "scope": "javascriptreact,typescriptreact",
+    "prefix": "component",
+    "body": [
+      "function ${1:Name}({ ${2:props} }) {",
+      "	return (",
+      "		${3:jsx}",
+      "	);",
+      "}"
+    ],
+    "description": "React Component boilerplate"
+  },
+  "iTwinUIComponent": {
+    "scope": "typescriptreact",
+    "prefix": "iuicomponent",
+    "body": [
+      "const ${1:Name} = React.forwardRef((props, forwardedRef) => {",
```

Comment:
This is missing imports for `React`, `PolymorphicForwardRefComponent` and `Box`. Is it possible to automate those?

Reply:
I'll look into it as well. Should be possible if it is possible to combine the two `cx` snippets.

Reply:
Decided to add a new `importForwardRef` snippet instead for handling all of these imports, as the second option on the previously link stack overflow page didn't work cleanly. There were two major issues, one being that the optimize imports command does not work on aliased imports like the `React` import, and the other being that imports were not actually moved to the top of the file by the command. I believe this issue was the result of the type declarations that often appear between our import block and where we declare our components.

Reply:
Similar to https://github.com/iTwin/iTwinUI/pull/1866#discussion_r1499641221, I would prefer a shorter prefix for the import. `iforwardRef` would be too weird though. Maybe `impforwardRef`?

You can play around with different names and see which one feels best.

Reply:
Alright, will do.

Reply:
I do like `impforwardRef` more overall, so I have changed the prefix to that. My only real concern overall is that we have two different import snippets, one that starts with `imp` and the other starts with just `i`. Would it be a good idea to rename the cx import prefix to `impcx`, just to create a consistent start to the import prefixes? Or should we just leave it as is for now?

Reply:
Sure, `impcx` is good with me

Reply:
Alright, changed in latest commit.